cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,659 @@
+        fig_re.show()
+        fig_imag.show()
+
+
+def rabi_oscillations(sampler: sim.SimulatesSamples, qubit: devices.GridQubit,
+                      max_angle: float = 2 * np.pi, *,
+                      repetitions: int = 1000,
+                      num_points: int = 200) -> RabiResult:
+    """Runs a Rabi oscillation experiment.
+
+    Rotates a qubit around the x-axis of the Bloch sphere by a sequence of Rabi
+    angles evenly spaced between 0 and max_angle. For each rotation, repeat
+    the circuit a number of times and measure the average probability of the
+    qubit being in the |1> state.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubit: The qubit under test.
+        max_angle: The final Rabi angle in radians.
+        repetitions: The number of repetitions of the circuit for each Rabi
+            angle.
+        num_points: The number of Rabi angles.
+
+    Returns:
+        A RabiResult object that stores and plots the result.
+    """
+    theta = sympy.Symbol('theta')
+    circuit = circuits.Circuit.from_ops(ops.X(qubit) ** theta)
+    circuit.append(ops.measure(qubit, key='z'))
+    sweep = study.Linspace(key='theta', start=0.0, stop=max_angle / np.pi,
+                           length=num_points)
+    results = sampler.run_sweep(circuit, params=sweep, repetitions=repetitions)
+    angles = np.linspace(0.0, max_angle, num_points)
+    excited_state_probs = np.zeros(num_points)
+    for i in range(num_points):
+        excited_state_probs[i] = np.mean(results[i].measurements['z'])
+
+    return RabiResult(angles, excited_state_probs)
+
+
+def single_qubit_randomized_benchmarking(sampler: sim.SimulatesSamples,
+                                         qubit: devices.GridQubit,
+                                         num_cliffords: Sequence[int],
+                                         use_xy_basis: bool = True,
+                                         *, num_circuits: int = 20,
+                                         repetitions: int = 1000
+                                         ) -> RandomizedBenchMarkResult:
+    """Clifford-based randomized benchmarking (RB) of a single qubit.
+
+    A total of num_circuits random circuits are generated, each of which
+    contains a fixed number of single-qubit Clifford gates plus one
+    additional Clifford that inverts the whole sequence and a measurement in
+    the z-basis. Each circuit is repeated a number of times and the average
+    |0> state population is determined from the measurement outcomes of all
+    of the circuits.
+
+    The above process is done for different numbers of Cliffords specified in
+    num_cliffords.
+
+    See Barends et al., Nature 508, 500 for details.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubit: The qubit under test.
+        num_cliffords: The different numbers of Cliffords in the RB study.
```

Comment:
Describe more generally why one would want to change this `num_cliffords` value, and what valid values look like. Can we pick a default that works well? Repeat for all methods with this argument.

It seems that this is setting the circuit sizes that will be sampled from, right? In that case I would try to use a different name, because `num_*` sounds like it should be an integer but this is a sequence of integers. Maybe `clifford_counts`? `lengths`?

Oh, maybe `num_clifford_range`! A `range(5, 20)` argument works right?

Reply:
Thanks for the suggestion here. `num_cliffords` is now `num_clifford_range` with a default value. The doc string has been expanded to provide some guideline to the user on how to choose `num_clifford_range` judiciously.